gc/base/SparseVirtualMemory.cpp

```diff
@@ -138,7 +146,8 @@ MM_SparseVirtualMemory::allocateSparseFreeEntryAndMapToHeapObject(void *proxyObj
 #endif /* defined(OSX) || defined(OMRZTPF)) */
 
 	if (NULL != sparseHeapAddr) {
-		_sparseDataPool->mapSparseDataPtrToHeapProxyObjectPtr(sparseHeapAddr, proxyObjPtr, adjustedSize);
+//		_sparseDataPool->mapSparseDataPtrToHeapProxyObjectPtr(sparseHeapAddr, proxyObjPtr, adjustedSize);
+		_sparseDataPool->mapSparseDataPtrToHeapProxyObjectPtr(sparseHeapAddr, proxyObjPtr, size);
```

Comment:
put a comment like:
while the allocate and commit will work with _pageSize aligned memory, the map will contain exact size of the object (so we can later verify the size on updates)

Reply:
BTW, I'm not sure what _pageSize is. By the original design, Offheap should be logically split into regions of the same size as the main Heap, while I suspect that _pageSize might be a physical page size....

Reply:
FYI: Sparse Heap Page size is controlled by `extensions->sparseHeapPageSize`. Currently it happen to be initialized as element 0 of page sizes array (small pages) and never changed since. There is no even command line option to set is explicitly. So, it is 4k for all platforms except Linux PPCLE where it can be 64k.

Reply:
I am not sure we should match Object Heap and Sparse Heap pages. Pre-reserved Linux Huge pages can not be recommitted, so it might be source of extended memory usage.